src/pdf.js

```diff
@@ -30,11 +30,26 @@ var pdfjsDisplaySVG = require("./display/svg.js");
 let pdfjsDisplayWorkerOptions = require("./display/worker_options.js");
 let pdfjsDisplayAPICompatibility = require("./display/api_compatibility.js");
 
-if (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) {
+if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("PRODUCTION")) {
+  pdfjsDisplayAPI.setPDFNetworkStreamFactory(async params => {
+    const [{ PDFNetworkStream }, { PDFFetchStream }] = await Promise.all([
+      SystemJS.import("../src/display/network.js"),
```

Comment:
It's not entirely clear to me why this new code block is necessary since it adds new `SystemJS` usage. I can see that we're now not running the `require` statement for the development viewer case anymore, but since it's being exchanged by direct `SystemJS` usage I'm not sure what the advantage is here?

Reply:
The main idea in this PR was to reduce usage of `require` in code which runs in development mode, and the idea here is that we'll be able to easily converted these ones to native [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports) later.

Another "fun" follow-up that will be necessary is to ensure that *all* import statements in the entire code-base have a `.js` extension, since otherwise the native `import` will fail with MIMI type errors...